File: deck.java
import java.util.ArrayList;
import java.util.Collections;
import java.util.List;

public class Deck {
    private List<Card> cards;

    public Deck() {
        cards = new ArrayList<>();
        String[] suits = {"♥", "♦", "♠", "♣"};
        String[] ranks = {"A", "2", "3", "4", "5", "6", "7", "8", "9", "10", "J", "Q", "K"};

        for (String s : suits) {
            for (String r : ranks) {
                cards.add(new Card(s, r));
            }
        }
    }

    public void shuffle() {
        Collections.shuffle(cards);
    }

    public Card draw() {
        if (cards.isEmpty()) return null;
        return cards.remove(cards.size() - 1);
    }
}

File: js版/コード/Deck.js
/**
 * Deckクラス (JavaのDeck.javaの移植版)
 */
class Deck {
    constructor() {
        this.cards = [];
        // 🏆 修正：画像連番のルールに合わせて並び順を変更しました
        const suits = ["♠", "♣", "♦", "♥"];
        const ranks = ["A", "2", "3", "4", "5", "6", "7", "8", "9", "10", "J", "Q", "K"];

        for (let s of suits) {
            for (let r of ranks) {
                this.cards.push(new Card(s, r));
            }
        }
    }

    // JavaのCollections.shuffleに相当
    shuffle() {
        for (let i = this.cards.length - 1; i > 0; i--) {
            const j = Math.floor(Math.random() * (i + 1));
            [this.cards[i], this.cards[j]] = [this.cards[j], this.cards[i]];
        }
    }

    draw() {
        return this.cards.pop();
    }
}